packages/patternfly-4/react-core/src/components/Chip/Chip.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ChipButton from './ChipButton';
+import ChipText from './ChipText';
+import ChipWrapper from './ChipWrapper';
+import { TimesCircleIcon } from '@patternfly/react-icons';
+
+export const ChipVariant = {
+  overflow: 'overflow',
+  closable: 'closable'
+};
+
+const Chip = ({ variant, onClick, children, id, className, ...props }) => {
+  switch (variant) {
+    case ChipVariant.overflow:
+      return (
+        <ChipWrapper isOverflowed className={className} {...props}>
+          <ChipButton onClick={onClick} aria-label={children}>
+            <ChipText>{children}</ChipText>
+          </ChipButton>
+        </ChipWrapper>
+      );
+    default:
+      return (
+        <ChipWrapper className={className} {...props}>
+          <ChipText id={id}>{children}</ChipText>
+          <ChipButton onClick={onClick} aria-label="Remove" id={`remove_${id}`} aria-labelledby={`remove_${id} ${id}`}>
+            <TimesCircleIcon aria-hidden="true" />
+          </ChipButton>
+        </ChipWrapper>
+      );
+  }
+};
+
+Chip.propTypes = {
+  /** Content rendered inside the chip text */
+  children: PropTypes.string.isRequired,
+  /** ID of the chip */
+  id: props => {
```

Comment:
In other components, if the consumer does not provide an id that is required for accessibility,  we  have been generating it internally.  Maybe we should consider doing that here and getting rid of this check.